### ISSUE TITLE
Guide: Be clearer about the default refinement behaviour and how refine is working

### ISSUE THREAD

**Issue by mthuret** [algolia/react-instantsearch#69]
See #61. 

We should explain in our documentation what to expect when clearing refinements with default refinements (ie: they are not removed).

In the same time it would be nice to check for every connector how their `refine` function works, to set or remove a refinement. It should be consistent, but today it's not (for instance: `connectRange`). 

We should wrote a guide about those results.

**Comment by davidfurlong** [comment]
Thanks for moving the issue. I'd like to add that currently removing the refinement for `connectRange` using refine seems to be impossible once set - `refine(null)` and `refine({min: null, max: null})` both don't work. It may even make more sense to provide another function in the props which removes a refinement, rather than overloading the `refine` function

**Comment by davidfurlong** [comment]
Are `defaultRefinement`s cleared on a clearAll? My code seems to behave as such.

**Comment by samouss** [comment]
Yes the `defaultRefinement` are clear when `ClearAll` is clicked.

As a workaround you can filter the items to clear with the `transformItems` prop. In the following example all the current refinement are removed, except the `category` of the `VirtualMenu`.

```jsx
<InstantSearch
  appId="latency"
  apiKey="6be0576ff61c053d5f9a3225e2a90f76"
  indexName="ikea"
>
  <VirtualMenu 
    attributeName="category" 
    defaultRefinement="Dining" 
  />

  <RefinementList 
    attributeName="materials" 
    defaultRefinement={["Polyamide"]} 
  />

  <ClearAll
    transformItems={items => items.filter(item => item.attributeName !== "category")}
  />

  <Hits />
</InstantSearch>
```

**Comment by njbarrett** [comment]
@samouss thanks that's exactly what I was looking for

**Comment by albertojacini** [comment]
With > v5 use item.attribute instead of item.attributeName in : 
`  <ClearAll
    transformItems={items => items.filter(item => item.attributeName !== "category")}
  />`

**Comment by FabienMotte** [comment]
Hey!

We're doing a round of clean up before migrating this repository to the new [InstantSearch monorepo](https://github.com/algolia/instantsearch.js). This issue seems not to have generated much activity lately and to be mostly solved by the shared code snippet, so we're going to close it.

Here's a sandbox using React InstantSearch Hooks: https://codesandbox.io/s/upbeat-forest-hpnq58?file=/src/App.tsx

To clear the active refinements, you can now use the [`<ClearRefinements>`](https://www.algolia.com/doc/api-reference/widgets/clear-refinements/react-hooks/) widget with React InstantSearch Hooks. You can also exclude attributes from the refinements to clear using the [`excludedAttributes`](https://www.algolia.com/doc/api-reference/widgets/clear-refinements/react-hooks/#widget-param-excludedattributes) prop.

Default refinements are now defined by passing an [`initialUiState`](https://www.algolia.com/doc/api-reference/widgets/clear-refinements/react-hooks/#widget-param-excludedattributes) prop on the `<InstantSearch>` component.
As you can see on the sandbox, default refinements are cleared by the `<ClearRefinements>` widget (unless they are excluded).

We recommend anyone who is currently using React InstantSearch v6 to [migrate to Hooks](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react-hooks/).
